### ISSUE TITLE
Add support to run separate command only once after first watch iteration

### ISSUE THREAD

**Issue by malteneuss** [watchexec/watchexec#456]
**Is your feature request related to a problem? Please describe.**
I'm using watchexec around the diagram tool plantuml that converts `test.puml` code into an `test.png` image as follows
`watchexec -f test.puml  plantuml test.puml` (super convenient). However, in order to see that image i currently have to *separately* open that image in a (non-flashing) image viewer like `feh test.png` or `ristretto test.png`, which from then on can auto-reload on any image file changes.

**Describe the solution you'd like**
It would be nice if i could open such an image viewer within that watchexec call once, namely after the first iteration; before that the converted image may not exist yet; after the first iteration the auto-reload feature of the image viewer takes over. For example, with a new option `--once '<command to once run after first iteration>'`:
```
watchexec --once 'feh test.png' -f test.puml  plantuml test.puml
```

This way we could run this and similar workflows in one simple terminal command.

**Describe alternatives you've considered**
I've tried starting the image viewer in parallel 
`feh test.png &; watchexec -f test.puml plantuml test.puml`. However, the image viewer won't start before the first watchexec iteration has finished, because that converted image doesn't yet exist.

I've tried restarting the image viewer with every watchexec iteration
`watchexec -f test.puml 'echo test && plantuml test.puml && feh --reload test.png &'`
but this just keeps opening additional image viewer instances.

**Additional context**
I haven't found any other file watch tool that could do that.

**Comment by passcod** [comment]
That seems a bit too specialised for the current state of watchexec, though not inconceivable in the future. Critically, your usecase is particular about when the `once` command should run (after the main command has finished running once), but I can see a bunch of other times that could be requested on the back of that, e.g. before the main command on the first run, in parallel, what happens on failure or on quit, etc.

One workaround I can think of is to have a somewhat more complex script and check some state. In pseudo code:
- if not file exists `test.png`
  - convert
  - start feh and detach it
- else
  - convert only

Another is to operate outside of watchexec:
- Create an empty or trivial png
- Start feh pointing to it
- Start watchexec which will overwrite the png

**Comment by malteneuss** [comment]
Thanks for your quick feedback. I see that the convenience gain is small (although watchexec is already extremely convenient compared to similar tools like `encr`). However, there are still many converters without a watch feature, which benefit from using `watchexec` for live editing in the way i described. Would it be okay to keep this thread open for a while to get feedback if others would like such a feature? 

Also could i add an example of this use case to the docs similar to your pseudo code? Something like
`SOURCE="test.puml"; TARGET="test.png"; watchexec -f $SOURCE "if [ -f $TARGET ]; then plantuml $SOURCE && feh $TARGET; else plantuml $SOURCE; fi"` or with pandoc.

**Comment by passcod** [comment]
Yeah, I'll keep this open. I could see this being possible natively in the future, too (..."always in motion is the future").

**Comment by passcod** [comment]
And if you can clean the example up a bit (multiline, add an explanation), then please PR it to the docs yes :)

**Comment by malteneuss** [comment]
Great, i added a PR with a cleaned up, explained version of the example.
